app/src/main/java/org/fossasia/phimpme/SharingActivity.java

```diff
@@ -272,15 +273,30 @@ private void otherShare() {
     }
 
     private void shareToInstagram() {
-        caption = text_caption.getText().toString();
-        Uri uri = Uri.fromFile(new File(saveFilePath));
-        Intent share = new Intent(Intent.ACTION_SEND);
-        share.setPackage("com.instagram.android");
-        share.putExtra(Intent.EXTRA_STREAM, uri);
-        share.setType("image/*");
-        share.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
-        startActivity(Intent.createChooser(share, caption));
-        atleastOneShare = true;
+        if(isAppInstalled("com.instagram.android")) {
+            caption = text_caption.getText().toString();
+            Uri uri = Uri.fromFile(new File(saveFilePath));
+            Intent share = new Intent(Intent.ACTION_SEND);
+            share.setPackage("com.instagram.android");
+            share.putExtra(Intent.EXTRA_STREAM, uri);
+            share.setType("image/*");
+            share.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+            startActivity(Intent.createChooser(share, caption));
+            atleastOneShare = true;
+        }else
+            Snackbar.make(parent,R.string.instagram_not_installed,Snackbar.LENGTH_LONG).show();
+    }
+
+    private boolean isAppInstalled(String packageName) {
```

Comment:
Move this function in `Utils` class, because its generic and if we require  to use this in future, Don't need to create again.